bitcoin/src/network/mod.rs

```diff
@@ -57,27 +57,75 @@ impl From<Network> for NetworkKind {
 
         match n {
             Bitcoin => NetworkKind::Main,
-            Testnet | Signet | Regtest => NetworkKind::Test,
+            Testnet(_) | Signet | Regtest => NetworkKind::Test,
         }
     }
 }
 
-/// The cryptocurrency network to act on.
+/// The testnet version to act on.
 #[derive(Copy, PartialEq, Eq, PartialOrd, Ord, Clone, Hash, Debug)]
 #[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
 #[cfg_attr(feature = "serde", serde(rename_all = "lowercase"))]
 #[non_exhaustive]
```

Comment:
In ae178a392f9774f31195aff3b49830e9c00fb3e7:

I don't think we should have serde impls on the `TestnetVersion` type. In general I think we should avoid adding serde impls on new types unless we have a clear usecase.

Reply:
I'd say that `TestnetVersion` is obviously an integer but I agree we should stay conservative.